### ISSUE TITLE
triangle problem

### ISSUE THREAD

**Issue by mdsumner** [r-gris/rangl#28]
This doesn't work, a certain combination of polygons and max_area - the linking with vindex$row_n is getting out of whack somehow, is it the order of the verts or is there somehow a too-small triangle throwing it out?

```R
## problem
p <- sp::spTransform(wrld_simpl, "+proj=laea +ellps=WGS84")

edge <- rangl(p[seq(2, nrow(p), by= 8), ], max_area = 5e8)
plot(edges$v$x_, edges$v$y_, type = "n")
## the rangl tables are ... not fine
## because this plot is ... not fine
lapply(split(edge$tXv, edge$tXv$triangle_), function(x) lines((inner_join(x, edge$v) )[, c("x_", "y_")], col = "firebrick"))
#Joining, by = "vertex_"
#Error: not compatible with requested type 

## the rgl constructs are now bogus
## (the plot is true, the indexes are out of whack some how)
a <- plot(edge)
g <- plot(globe(edge))
ncol(a$vb)
ncol(a$it)
```

**Comment by mdsumner** [comment]
Nope, don't know. This sequence triggers it, but not sure which subsets of the sequence do

```R

library(maptools)
data(wrld_simpl)
library(dplyr)
## problem
p <- sp::spTransform(wrld_simpl, "+proj=laea +ellps=WGS84")
library(rangl)

edge <- rangl(p[seq(20, 40, by = 1), ], max_area = 5e8)
rgl.clear()
a <- plot(edge)

```

**Comment by mdsumner** [comment]
Similar with ssmu[1:14, ] is ok but afer that goes haywire

**Comment by mdsumner** [comment]
This is fixed by relying on the in-built behaviour of duplicated, rather than dplyr::distinct(). Fixed in 8b3d9ec

**Comment by mdsumner** [comment]
ssmu has some very large number of vertices, and this is a different problem - rows 14 and 17 are big because of the baked in coastline (100K and 220K + vertices).
